neo/public/javascripts/cache/services.js

```diff
@@ -0,0 +1,356 @@
+ angular.module('cloudberry.cache', ['leaflet-directive', 'cloudberry.common' ])
+ .service('Cache', function( $window, $http, $compile, Asterix){
+
+
+  var cachedCityPolygonTree = rbush();
+  var cachedRegion ;
+  var cacheSize = 0;
+  var insertedTreeIDs = new Set();
+  var cacheThreshold = 5500;//soft limit
+  var DeleteTarget = 0;
+  var DeletedCount = 0;
+  var preFetchDistance = 25;
+
+
+
+  this.getCityPolygonsFromCache = function city(bounds){
+
+       var deferred = new $.Deferred();
+       var data_response;
+
+       var rteBounds = "city/" + bounds._northEast.lat + "/" + bounds._southWest.lat + "/" + bounds._northEast.lng + "/" + bounds._southWest.lng;
+
+       var currentRequestPolygon = turf.polygon([[
+                                    [bounds._northEast.lng,bounds._northEast.lat],
+                                    [bounds._northEast.lng,bounds._southWest.lat],
+                                    [bounds._southWest.lng,bounds._southWest.lat],
+                                    [bounds._southWest.lng,bounds._northEast.lat],
+                                    [bounds._northEast.lng,bounds._northEast.lat]
+                                ]]);
+
+       var bbox = turf.bbox(currentRequestPolygon);
+        // to search in Rbush Tree we need this in this format
+       var item = {
+           minX : bbox[0],
+           minY : bbox[1],
+           maxX : bbox[2],
+           maxY : bbox[3]
+       }
+
+
+       if(cachedRegion != undefined && turf.difference(currentRequestPolygon,cachedRegion) == undefined) {
```

Comment:
I think this is a very unsafe way of comparison. you may want to simply 
```
if(!cachedRegion  && turf.difference(currentRequestPolygon,cachedRegion) )
```
otherwise, you should use `===` which usually causes more problems.